File: OpenMP/solutions/omp_private.f
      program private

        use omp_lib
        implicit none

        integer :: tid, nthreads

c       Fork a team of threads

!$OMP PARALLEL PRIVATE(tid)

        tid = omp_get_thread_num();
        nthreads = omp_get_num_threads();
        print *,"Hello World from thread = ",tid," out of ",nthreads

!$OMP END PARALLEL

      end program

File: OpenMP/solutions/omp_private.c
#include <omp.h>
#include <stdio.h>

int main (int argc, char *argv[])
{

int tid, nthreads;

/* Fork a team of threads */
#pragma omp parallel private(tid)
  {
  tid = omp_get_thread_num();
  nthreads = omp_get_num_threads();
  printf("Hello World from thread = %d out of %d\n", tid, nthreads);
  }  /* All threads join master thread and disband */
}
